File: frontend/keepnot/lib/data/models/checklist_item.dart
class ChecklistItem {
  final String? id;
  final String noteId;
  final String itemText;
  final bool isDone;

  ChecklistItem({
    this.id,
    required this.noteId,
    required this.itemText,
    this.isDone = false,
  });

  Map<String, dynamic> toMap() => {
    'note_id': noteId,
    'item_text': itemText,
    'is_done': isDone,
  };

  factory ChecklistItem.fromMap(Map<String, dynamic> map, String docId) =>
      ChecklistItem(
        id: docId,
        noteId: map['note_id'] ?? '',
        itemText: map['item_text'] ?? '',
        isDone: map['is_done'] ?? false,
      );
}

File: backend/GeoNotes/src/main/java/com/ydg/geonotes/entity/ChecklistItem.java
package com.ydg.geonotes.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

@Entity
@Table(name="checklist_items")
public class ChecklistItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String itemText;
    private boolean isDone;

    @ManyToOne
    @JoinColumn(name= "note_id")
    @JsonIgnore
    private Note note;

    public ChecklistItem() {}

    public ChecklistItem(Long id, String itemText, boolean isDone, Note note) {
        this.id = id;
        this.itemText = itemText;
        this.isDone = isDone;
        this.note = note;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getItemText() {
        return itemText;
    }

    public void setItemText(String itemText) {
        this.itemText = itemText;
    }

    public boolean isIsDone() {
        return isDone;
    }

    public void setIsDone(boolean done) {
        isDone = done;
    }

    public Note getNote() {
        return note;
    }

    public void setNote(Note note) {
        this.note = note;
    }
}
